crafting-dead-core/src/main/java/com/craftingdead/core/world/explosion/ConfigurableExplosion.java

```diff
@@ -0,0 +1,133 @@
+package com.craftingdead.core.world.explosion;
+
+import com.google.common.collect.Sets;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import net.minecraft.core.BlockPos;
+import net.minecraft.util.Mth;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.item.PrimedTnt;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.enchantment.ProtectionEnchantment;
+import net.minecraft.world.level.Explosion;
+import net.minecraft.world.level.ExplosionDamageCalculator;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.gameevent.GameEvent;
+import net.minecraft.world.level.material.FluidState;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.Vec3;
+import org.jetbrains.annotations.Nullable;
+
+public class ConfigurableExplosion extends Explosion {
```

Comment:
I'm not sure if you're familiar with Mixin but this would be much nicer as a Mixin instead of creating a new class and access transforming all the fields.

Reply:
Add a check in the Mixin to see if the source entity is a grenade